### PR TITLE
Run clippy on CI

### PR DESCRIPTION
**Pull request by pitbuster**
First take on #287. I am not sure if running nightly `clippy` is a requirement for the project (in my experience it is quite unstable), but we can work on that if that's the case.

### CODE REVIEW

File: .github/workflows/clippy.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+jobs:
+  lints:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Rust toolchain and cache
+        uses: actions-rust-lang/setup-rust-toolchain@v1.5.0
```
Comment (waywardmonkeys) [inline .github/workflows/clippy.yaml:17]: If this is added, it should use the same stuff for managing the toolchain as what's in the other CI jobs.

File: .github/workflows/clippy.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        with:
+          rustflags: ""
+
+      - name: Clippy
+        run: cargo clippy -- --all-features -Dwarnings
```
Comment (waywardmonkeys) [inline .github/workflows/clippy.yaml:22]: This should probably be checking all targets in the workspace.

**Review by waywardmonkeys — CHANGES_REQUESTED** [review]
I'm not sure if this should be a new job or just part of the existing test workflow.

File: .github/workflows/clippy.yaml (hunk 2 above)
Comment (pitbuster) [reply .github/workflows/clippy.yaml:22]: It should already be doing that.

**Comment by alerque** [comment]
Personally (having setup and managed a lot of CI stuff) I would probably not put this in the same workflow as the test and coverage jobs. I would either keep it completely separate like it is or adapt the existing `format.yml` workflow to include all linting related jobs. There isn't a huge difference between these options. Some tooling (like external badge services) make it a little easier to get the status of a workflow rather than a single job in a milti-job workflow, but it isn't a big deal.

File: .github/workflows/clippy.yaml (hunk 1 above)
Comment (alerque) [reply .github/workflows/clippy.yaml:17]: Yes, definitely. This needs to use the same Rust setup tooling as the other workflows. If there is a reason to change that (which their might be but that should be weighed in a different issue/PR) we should change all of them to be consistent.

File: .github/workflows/clippy.yaml (hunk 2 above)
Comment (alerque) [reply .github/workflows/clippy.yaml:22]: I confirmed it is already checking all the members (except the one that is currently excluded and known-broken, that's a matter for a different issue).

File: .github/workflows/clippy.yaml (hunk 1 above)
Comment (pitbuster) [reply .github/workflows/clippy.yaml]: changed it to use the same action as the other workflows

**Review by waywardmonkeys — APPROVED** [review]

**Pull request merged**